### PR TITLE
Start writing unit tests

### PR DESCRIPTION
**Pull request by leplatrem**
- [x] Tests setup (Mocha+Browser+Travis)
- [x] Moch daybed server requests
- [x] Basic tests for expected requests and responses

### CODE REVIEW

**Comment by Natim** [comment]
Great :)

**Comment by leplatrem** [comment]
Done :) The scope of my pull-request was to setup tests and mocking server requests.
If you're ok with that, you can merge :)

File: index.js [hunk 1]
```diff
@@ -30,10 +30,10 @@
       }
       req.onload = function() {
         if (!("" + req.status).match(/^2/)) {
-          reject(req.response);
+          reject(new Error(req.responseText));
           return;
         }
-        resolve(req.response);
+        resolve(JSON.parse(req.responseText));
       };
```
Comment (leplatrem) [inline index.js:37]: Since we force the accept mime-type, it is ok I guess

File: index.js [hunk 2]
```diff
@@ -30,10 +30,10 @@
       }
       req.onload = function() {
         if (!("" + req.status).match(/^2/)) {
-          reject(req.response);
+          reject(new Error(req.responseText));
```
Comment (leplatrem) [inline index.js:33]: responseText is the raw stuff

File: index.js (hunk 1 above)
Comment (leplatrem) [inline index.js:37]: _notes_: Mocking XMLHttpRequest with sinon did not work well with `req.responseType` and `req.response`

**Comment by Natim** [comment]
Great thanks a lot.

**Pull request merged**